src/process_utility.c

```diff
@@ -1324,19 +1324,28 @@ process_altertable_end_subcmd(Hypertable *ht, Node *parsetree, ObjectAddress *ob
 	switch (cmd->subtype)
 	{
 		case AT_ChangeOwner:
-			process_altertable_change_owner(ht, cmd);
-			break;
-		case AT_AddIndexConstraint:
-			ereport(ERROR,
-					(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
-					 errmsg("Hypertables currently do not support adding "
-							"a constraint using an existing index.")));
+			if (NULL != ht)
+				process_altertable_change_owner(ht, cmd);
 			break;
+		case AT_AddIndexConstraint: 
+			{
+				if (NULL == ht)
```

Comment:
Hmm if all the cases check for whether `ht` is `NULL` (and I _think_ they should), then we probably just want to wrap the whole switch statement in an `if (NULL != ht)`, right?

Reply:
This was the case prior to this pr; however we may want to address when a client issues something like the the following statement:

`ALTER TABLE non_hypertable_partitioned ATTACH PARTITION tempurature_hypertable`

In this case, `ht` will be null since non_hypertable_partitioned is not a hypertable. The AT_AttachPartition case will check to see if the attaching partition is a hypertable. For all other cases, I preserved the original functionality to disregard events on tables that were not hypertables.

Reply:
Good point, makes sense then.